Added Kepler Energy Sources doc (#76)
sustainable-computing-io/kepler-doc: docs/design/kepler-energy-sources.md

The MSRs can be accessed directly on Linux using the msr driver in the kernel. Reading RAPL domain values directly from MSRs requires detecting the CPU model and reading the RAPL energy units before reading the RAPL domain. Once the CPU model is detected, the RAPL domains can be read per package of the CPU by reading the corresponding ’MSR status’ register. 

There are basically two types of events that RAPL events report
Static Events: thermal specifications, maximum and minimum power caps, and time windows.
Dynamic Events: RAPL domain energy readings from the chip such as PKG, PP0, PP1 or DRAM 

### Using RAPL Sysfs 
From Linux Kernel version 3.13 onwards, RAPL values can be read using “Power Capping Framework” [2].  

Linux Power Capping framework exposes power capping devices to user space via sysfs in the form of a tree of objects. 

> This sysfs tree is mounted at /sys/devices/virtual/powercap

Reviewer: ```suggestion
This sysfs tree is mounted at /sys/class/powercap/intel-rapl
```

Author: i mentioned `/sys/devices/virtual/powercap` because this is what is mentioned in Kernel documentation
https://www.kernel.org/doc/html/next/power/powercap/powercap.html

Later i have mentioned that 
```
Kepler reads path /sys/class/powercap which contains symlinks to above the sysfs folder.
```

> Kepler reads path /sys/class/powercap which contains symlinks to above the sysfs folder.

Reviewer: ```suggestion
Kepler reads path /sys/class/powercap/intel-rapl which contains sub folders that container RAPL domain data.
```

### Using kernel driver xgene-hwmon
Using Xgene-hwmon driver kepler reads power from APM X-Gene SoC. It supports reading CPU and IO power in micro watts. 

### Using eBpf perf events 
Not used in kepler

### Using PAPI library
Performance Application Programming Interface (PAPI) 
Not used in kepler

Kepler chooses to use one enenry sourcex in the following order of preference
1. Sysfs
2. MSR
3. Hwmon

> 4. Static Power Estimated based on time and num of cpu, size of DRAM

Reviewer: ```suggestion

```

Author: do we delete this line? or can replace with better explanation?

Reviewer: yes, please delete it, the static estimate is not used any more.
